src/main/java/com/salesforce/comdagen/model/Content.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2018, salesforce.com, inc.
+ *  All rights reserved.
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.comdagen.model
+
+import com.salesforce.comdagen.RandomData
+import com.salesforce.comdagen.SupportedZone
+import com.salesforce.comdagen.config.ContentConfiguration
+
+
+/**
+ * Abstract content class which requires a seed and a contentConfig.
+ */
+abstract class AbstractContent(
+    open val seed: Long,
+    private val contentConfig: ContentConfiguration
+) {
+    open val attributeId: String get() = contentConfig.attributeId
+    open val contentId: String
+        get() = contentConfig.contentId ?: "unnamed-content-id"
```

Comment:
Aren't content ids unique per library? So this fallback would end up creating non-functional assets.